[BUG] Shimplify adding new shim layer fails

**Describe the bug**
Trying to add a new shim layer for Spark 3.3.3 I ran into an error running:

```
mvn generate-sources -Dshimplify=true \
    -Dshimplify.move=true -Dshimplify.overwrite=true \
    -Dshimplify.add.shim=333 -Dshimplify.add.base=332
```

Error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (shimplify-shim-sources) on project rapids-4-spark-sql_2.12: A
n Ant BuildException has occured: Traceback (most recent call last):
[ERROR]   File "<script>", line 592, in <module>
[ERROR]   File "<script>", line 348, in task_impl
[ERROR]   File "<script>", line 481, in __shimplify_layout
[ERROR]   File "<script>", line 533, in __add_new_shim_to_file_map
[ERROR] AssertionError: Per-shim file are expect to belong to a single shim, actual number of shims: 2
```

It looks like the issue is https://github.com/NVIDIA/spark-rapids/blob/branch-23.04/sql-plugin/src/main/spark331/scala/com/nvidia/spark/rapids/shims/SparkShims.scala has 2 shims in it.  Originally this is what we wanted but with other changes ( https://github.com/NVIDIA/spark-rapids/pull/7857) that have been made we should now be able to use SparkShims.scala across different versions of Spark.  

So we need to update the shimplify tool to remove this requirement.

--- a/build/shimplify.py
+++ b/build/shimplify.py
@@ -522,7 +522,7 @@ def __add_new_shim_to_file_map(files2bv):
         bv_list = files2bv[shim_file]
         if __add_shim_base in bv_list:
             # adding a lookalike
-            # case 1) dedicated per-shim files with a spark${buldver} in the package path:
+            # case 1) dedicated per-shim files with a spark${buildver} in the package path:
             #         CLONE the file with modifications
             # case 2) otherwise simply add the new buildver to the files2bv[shimfile] mapping
             base_package = "spark%s" % __add_shim_base

--- a/build/shimplify.py
+++ b/build/shimplify.py
@@ -522,15 +522,13 @@ def __add_new_shim_to_file_map(files2bv):
         bv_list = files2bv[shim_file]
         if __add_shim_base in bv_list:
             # adding a lookalike
-            # case 1) dedicated per-shim files such as SparkShims.scala and anything with
-            #         a spark${buldver} in the package path: CLONE the file with modifications
+            # case 1) dedicated per-shim files with a spark${buldver} in the package path:
+            #         CLONE the file with modifications
             # case 2) otherwise simply add the new buildver to the files2bv[shimfile] mapping
             base_package = "spark%s" % __add_shim_base
-            shim_impl_file = 'com.nvidia.spark.rapids.shims.SparkShims'\
-                .replace('.', os.sep) + '.scala'
-            if (base_package in shim_file) or shim_file.endswith(shim_impl_file):
-                assert len(bv_list) == 1, "Per-shim file are expect to belong to a single "\
-                        "shim, actual number of shims: %s" % len(bv_list)
+            if base_package in shim_file:
+                assert len(bv_list) == 1, "Per-shim file %s is expected to belong to a single "\
+                        "shim, actual shims: %s" % (shim_file, bv_list)
                 new_shim_file = __git_rename_or_copy(shim_file, __add_shim_buildver,
                                                      from_shim=__add_shim_base)
                 # schedule new file for comment update

